OSDOCS#12867: Docs for hibernating a cluster (#86873)
openshift/openshift-docs: modules/hibernating-cluster-resume.adoc

// Module included in the following assemblies:
//
// * backup_and_restore/hibernating-cluster.adoc

:_mod-docs-content-type: PROCEDURE
[id="hibernating-cluster-resume_{context}"]
= Resuming a hibernated cluster

> You can resume a hibernated cluster within 90 days. You might have to approve certificate signing requests (CSRs) for the nodes to become ready.

Reviewer: ```suggestion
When you resume a hibernated cluster within 90 days, during this period, you might have to approve certificate signing requests (CSRs) for the nodes to become ready.
```

It can take around 45 minutes for the cluster to resume, depending on the size of your cluster.

.Prerequisites

* You hibernated your cluster less than 90 days ago.
* You have access to the cluster as a user with the `cluster-admin` role.

.Procedure

> . Within 90 days of cluster hibernation, resume the cluster virtual machines:
> +
> Use the tools native to your cluster's cloud environment to resume the cluster's virtual machines.

Reviewer 2: Same ? re: single para:
```suggestion
. Within 90 days of cluster hibernation, resume the cluster virtual machines. Use the tools native to your cluster's cloud environment to resume the cluster's virtual machines.
```
or no-colon:
```suggestion
. Within 90 days of cluster hibernation, resume the cluster virtual machines.
+
Use the tools native to your cluster's cloud environment to resume the cluster's virtual machines.
```

. Wait about 5 minutes, depending on the number of nodes in your cluster.

> . Approve certificate signing requests (CSRs) for the nodes:

Reviewer 2: Could drop long-form here since already defined in module:
```suggestion
. Approve CSRs for the nodes:
```

----

.. Approve each valid CSR by running the following command:
+
[source,terminal]
----
$ oc adm certificate approve <csr_name>
----

.. Verify that all necessary CSRs were approved by running the following command:
+
[source,terminal]
----
$ oc get csr
----
+
.Example output
[source,terminal]
----
NAME       AGE  SIGNERNAME                                   REQUESTOR                                                                  REQUESTEDDURATION  CONDITION
csr-4dwsd  37m  kubernetes.io/kube-apiserver-client          system:node:ci-ln-812tb4k-72292-8bcj7-worker-c-q8mw2                       24h                Approved,Issued
csr-4vrbr  49m  kubernetes.io/kube-apiserver-client          system:node:ci-ln-812tb4k-72292-8bcj7-master-1                             24h                Approved,Issued
csr-4wk5x  51m  kubernetes.io/kubelet-serving                system:node:ci-ln-812tb4k-72292-8bcj7-master-1                             <none>             Approved,Issued
csr-84vb6  51m  kubernetes.io/kube-apiserver-client-kubelet  system:serviceaccount:openshift-machine-config-operator:node-bootstrapper  <none>             Approved,Issued
----
+

> CSRs should show `Approved,Issued` in the CONDITION column.

Reviewer 2: ```suggestion
CSRs should show `Approved,Issued` in the `CONDITION` column.
```

. Verify that all nodes now show as ready by running the following command:
+
[source,terminal]
----
$ oc get nodes
----
+
.Example output
[source,terminal]
----
NAME                                      STATUS  ROLES                 AGE   VERSION
ci-ln-812tb4k-72292-8bcj7-master-0        Ready	  control-plane,master  32m   v1.31.3
ci-ln-812tb4k-72292-8bcj7-master-1        Ready	  control-plane,master  32m   v1.31.3
ci-ln-812tb4k-72292-8bcj7-master-2        Ready	  control-plane,master  32m   v1.31.3
Ci-ln-812tb4k-72292-8bcj7-worker-a-zhdvk  Ready	  worker                19m   v1.31.3
ci-ln-812tb4k-72292-8bcj7-worker-b-9hrmv  Ready	  worker                19m   v1.31.3
ci-ln-812tb4k-72292-8bcj7-worker-c-q8mw2  Ready	  worker                19m   v1.31.3
----
+

> All nodes should show `Ready` in the `STATUS` column. It might take a few moments for the nodes to become ready after approving the CSRs.

Reviewer: `a few moments ` is `a few minutes` ?

Author: Updated

> All nodes should show `Ready` in the `STATUS` column. It might take a few moments for the nodes to become ready after approving the CSRs.

Reviewer: ```suggestion
All nodes should show `Ready` in the `STATUS` column. It might take a few minutes for all nodes to become ready after approving the CSRs.
```

Author: Updated

// Module included in the following assemblies:
//
// * backup_and_restore/hibernating-cluster.adoc

:_mod-docs-content-type: PROCEDURE
[id="hibernating-cluster-resume_{context}"]
= Resuming a hibernated cluster

You can resume a hibernated cluster within 90 days. You might have to approve certificate signing requests (CSRs) for the nodes to become ready.

It can take around 45 minutes for the cluster to resume, depending on the size of your cluster.

.Prerequisites

* You hibernated your cluster less than 90 days ago.
+

> // TODO: What if they try to resume later than 90 days? What are the consequences?

Author: @vrutkovs Thoughts on whether we there are any consequences we should state on what happens if you leave a cluster hibernated for more than 90 days?

Reviewer 3: It varies a lot on the hibernation period.
If the resume time is more than 2 years, users would need to construct a kubeconfig to get access to the cluster. If its between 180 days and 2 years it may take longer for operators to recover (and possibly more manual steps to recover). We found no issues when the cluster is hibernated between 90 and 180 days.

Lets mention 90 days here as that's the period we committed to support

* You have access to the cluster as a user with the `cluster-admin` role.
+
// TODO: Confirm must be a cluster-admin

.Procedure

. Within 90 days of cluster hibernation, resume the cluster virtual machines:
+
Use the tools native to your cluster's cloud environment to resume the cluster's virtual machines.

. Wait about 5 minutes, depending on the number of nodes in your cluster.

> . TODO: Check node status. Sync w/ Vadim on possible script?

Author: @vrutkovs in the google doc you mentioned a possible script instead of checking the node status. Do you have an idea of what we should do here?

Reviewer 3: The script would be `oc --request-timeout=5s wait --for=condition=Progressing clusteroperator/kube-apiserver --timeout=300s` which waits for kube-apiserver operator to be progressing again. But we found out that its unreliable - see https://issues.redhat.com/browse/API-1614 - so for now we have to do with "wait for 5 minutes and approve CSRs as they appear"

Author: Okay thanks, I'll keep it as waiting and then making sure there's a CSR for each NotReady node

. Approve certificate signing requests (CSRs) for the nodes:

.. Check that there is a CSR for each node in the `NotReady` state:
+
[source,terminal]
----
$ oc get csr
----
+
.Example output
[source,terminal]
----
NAME       AGE  SIGNERNAME                                   REQUESTOR                                                                  REQUESTEDDURATION  CONDITION
csr-4dwsd  37m  kubernetes.io/kube-apiserver-client          system:node:ci-ln-812tb4k-72292-8bcj7-worker-c-q8mw2                       24h                Pending
csr-4vrbr  49m  kubernetes.io/kube-apiserver-client          system:node:ci-ln-812tb4k-72292-8bcj7-master-1                             24h                Pending
csr-4wk5x  51m  kubernetes.io/kubelet-serving                system:node:ci-ln-812tb4k-72292-8bcj7-master-1                             <none>             Pending
csr-84vb6  51m  kubernetes.io/kube-apiserver-client-kubelet  system:serviceaccount:openshift-machine-config-operator:node-bootstrapper  <none>             Pending
----

> .. Approve each valid CSR by running the following command:

Author: @vrutkovs I chose to err on the side of caution of approving CSRs one by one (and then providing the to do them all later on). Let me know if you think that's okay.

Reviewer 3: Good idea, that would be cleaner and equivalent to automated script

Author: @vrutkovs Do you think it's still worthwhile to show the alternative for approving all pending CSRs? Or should we just show only the one-by-one option?

Author: I ended up deciding to pull out the command to approve all of them.

+
[source,terminal]
----
$ oc get clusteroperators
----
+
.Example output
[source,terminal]
----
NAME                      VERSION                             AVAILABLE  PROGRESSING  DEGRADED  SINCE   MESSAGE
authentication            4.17.0-0.nightly-2024-12-19-020614  True       False        False     51m
baremetal                 4.17.0-0.nightly-2024-12-19-020614  True       False        False     72m
cloud-controller-manager  4.17.0-0.nightly-2024-12-19-020614  True       False        False     75m
cloud-credential          4.17.0-0.nightly-2024-12-19-020614  True       False        False     77m
cluster-api               4.17.0-0.nightly-2024-12-19-020614  True       False        False     42m
cluster-autoscaler        4.17.0-0.nightly-2024-12-19-020614  True       False        False     72m
config-operator           4.17.0-0.nightly-2024-12-19-020614  True       False        False     72m
console                   4.17.0-0.nightly-2024-12-19-020614  True       False        False     55m
...
----
+
All cluster Operators should show `AVAILABLE`=`True`, `PROGRESSING`=`False`, and `DEGRADED`=`False`.

// .Verification

> // TODO: Do we want to do the basic health checking? Or is that not really necessary?

Author: @vrutkovs The test cases had a few steps of basic heath checking (step 8). Just things like logging in, creating a new project and a new app. Do you think it's worthwhile to add those here, or should we just leave it after checking that nodes/cluster operators are good to go?

Reviewer 3: its not required for users to perform - we run these steps in CI to ensure operator status is correct
